organization/src/integrationTest/kotlin/com/cosmotech/organization/service/OrganizationServiceIntegrationTest.kt

```diff
@@ -1283,1447 +1283,1502 @@ class OrganizationServiceIntegrationTest : CsmRedisTestBase() {
-          assertEquals(ROLE_EDITOR, userACLRetrieved.role)
-        }
+        organizationApiService.updateOrganizationAccessControl(
+            organizationRegistered.id!!, TEST_USER_ID, OrganizationRole(role = ROLE_NONE))
       }
+    }
 
-      @Test
-      fun `updateOrganizationAccessControl with not owned organization, no WRITE_SECURITY permission, ROLE_NONE`() {
-        assertThrows<CsmClientException> {
-          val name = "o-connector-test-1"
-          runAsDifferentOrganizationUser()
-          val organizationRegistered =
-              organizationApiService.registerOrganization(createTestOrganization(name))
-
-          organizationApiService.addOrganizationAccessControl(
-              organizationRegistered.id!!,
-              OrganizationAccessControl(id = TEST_USER_ID, role = ROLE_VIEWER))
-
-          runAsPlatformAdmin()
+    @Test
+    fun `updateOrganizationAccessControl with owned organization and unknown ACL user`() {
+      assertThrows<CsmResourceNotFoundException> {
+        val name = "o-connector-test-1"
+        val organizationRegistered =
+            organizationApiService.registerOrganization(createTestOrganization(name))
 
-          organizationApiService.updateOrganizationAccessControl(
-              organizationRegistered.id!!, OTHER_TEST_USER_ID, OrganizationRole(role = ROLE_NONE))
-        }
+        organizationApiService.updateOrganizationAccessControl(
+            organizationRegistered.id!!, UNKNOWN_IDENTIFIER, OrganizationRole(role = ROLE_EDITOR))
       }
+    }
 
-      @Test
-      fun `removeOrganizationAccessControl with owned organization`() {
+    @Test
+    fun `updateOrganizationAccessControl with owned organization and wrong role`() {
+      assertThrows<CsmClientException> {
         val name = "o-connector-test-1"
         val organizationRegistered =
             organizationApiService.registerOrganization(createTestOrganization(name))
 
-        val otherUserACL = OrganizationAccessControl(id = OTHER_TEST_USER_ID, role = ROLE_VIEWER)
-        organizationApiService.addOrganizationAccessControl(
-            organizationRegistered.id!!, otherUserACL)
-
-        organizationApiService.removeOrganizationAccessControl(
-            organizationRegistered.id!!, OTHER_TEST_USER_ID)
-        assertThrows<CsmResourceNotFoundException> {
-          organizationApiService.getOrganizationAccessControl(
-              organizationRegistered.id!!, OTHER_TEST_USER_ID)
-        }
+        organizationApiService.updateOrganizationAccessControl(
+            organizationRegistered.id!!, TEST_USER_ID, OrganizationRole(role = UNKNOWN_IDENTIFIER))
       }
+    }
 
-      @Test
-      fun `removeOrganizationAccessControl with not owned organization, WRITE_SECURITY permission`() {
+    @Test
+    fun `updateOrganizationAccessControl with not owned organization, WRITE_SECURITY permission, can update user (!= ROLE_NONE)`() {
+      assertDoesNotThrow {
```

Comment:
assertDoesNotThrow should be specific else it's the same as not having it

Reply:
same comment as above